src/systems/velocity_control/VelocityControl.cc

```diff
@@ -86,15 +114,43 @@ void VelocityControl::Configure(const Entity &_entity,
     return;
   }
 
-  // Subscribe to commands
-  std::string topic{"/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel"};
+  // Subscribe to model commands
+  std::string modelTopic{"/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel"};
   if (_sdf->HasElement("topic"))
-    topic = _sdf->Get<std::string>("topic");
+    modelTopic = _sdf->Get<std::string>("topic");
+  modelTopic = transport::TopicUtils::AsValidTopic(modelTopic);
   this->dataPtr->node.Subscribe(
-    topic, &VelocityControlPrivate::OnCmdVel, this->dataPtr.get());
-
-  ignmsg << "VelocityControl subscribing to twist messages on [" << topic << "]"
+    modelTopic, &VelocityControlPrivate::OnCmdVel, this->dataPtr.get());
+  ignmsg << "VelocityControl subscribing to twist messages on ["
+         << modelTopic << "]"
          << std::endl;
+
+  // Ugly, but needed because the sdf::Element::GetElement is not a const
+  // function and _sdf is a const shared pointer to a const sdf::Element.
+  auto ptr = const_cast<sdf::Element *>(_sdf.get());
+
+  if (!ptr->HasElement("link_name"))
+    return;
+
+  sdf::ElementPtr sdfElem = ptr->GetElement("link_name");
+  while (sdfElem)
+  {
+    this->dataPtr->linkNames.push_back(sdfElem->Get<std::string>());
+    sdfElem = sdfElem->GetNextElement("link_name");
+  }
+
+  // Subscribe to link commands
+  for (const auto &linkName : this->dataPtr->linkNames)
+  {
+    std::string linkTopic{"/model/" + this->dataPtr->model.Name(_ecm) +
+                             "/link/" + linkName + "/cmd_vel"};
+    linkTopic = transport::TopicUtils::AsValidTopic(linkTopic);
```

Comment:
same here, check if the string is null

Reply:
added check. 468e2f68383b9188c10b1201fda73ac863eb8a42